pkg/build/graph/analysis/bc.go

```diff
@@ -68,6 +71,86 @@ bc:
 	return markers
 }
 
+// FindMissingInputImageStreams checks all build configs and confirms that their From element exists
+func FindMissingInputImageStreams(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+OUTER:
+	for _, bcNode := range g.NodesByKind(buildgraph.BuildConfigNodeKind) {
+		for _, istNode := range g.PredecessorNodesByEdgeKind(bcNode, buildedges.BuildInputImageEdgeKind) {
+			for _, uncastImageStreamNode := range g.SuccessorNodesByEdgeKind(istNode, imageedges.ReferencedImageStreamGraphEdgeKind) {
+				imageStreamNode := uncastImageStreamNode.(*imagegraph.ImageStreamNode)
+
+				// note, BuildConfig.Spec.BuildSpec.Strategy.[Docker|Source|Custom]Stragegy.From Input of ImageStream type will get converted to ImageStreamTag on the vX to api conversion
+
+				if !imageStreamNode.IsFound {
+					markers = append(markers, osgraph.Marker{
+						Node:         bcNode,
+						RelatedNodes: []graph.Node{istNode},
+
+						Severity: osgraph.ErrorSeverity,
+						Key:      MissingInputImageStreamErr,
+						Message: fmt.Sprintf("%s builds from %s, but the image stream does not exist.",
+							bcNode.(*buildgraph.BuildConfigNode).ResourceString(), imageStreamNode.ResourceString()),
+						Suggestion: osgraph.Suggestion(fmt.Sprintf("create the image stream %s with the required images", imageStreamNode.ResourceString())),
```

Comment:
$ oc describe build/ruby-sample-build-2
Name:		ruby-sample-build-2
Created:	About a minute ago
Labels:		app=ruby-sample-build,buildconfig=ruby-sample-build,name=ruby-sample-build,openshift.io/build-config.name=ruby-sample-build,template=application-template-stibuild
Annotations:	openshift.io/build.number=2
Build Config:	ruby-sample-build
Duration:	waiting for 1m13s
Build Pod:	ruby-sample-build-2-build
Strategy:	Source
URL:		https://github.com/openshift/ruby-hello-world.git
From Image:	DockerImage centos/ruby-22-centos7@sha256:72bc0354eca9ad377bd9d36ec7feabf8aaa1b76d38dfce10db08d7671e581429
Output to:	ImageStreamTag origin-ruby-sample:latest
Status:		New (The referenced output image stream default/origin-ruby-sample does not exist.)
Events:
  FirstSeen	LastSeen	Count	From			SubobjectPath	Reason			Message
  ─────────	────────	─────	────			─────────────	──────			───────
  1m		1m		1	{build-controller }			HandleBuildError	Build has error: the referenced output image stream default/origin-ruby-sample does not exist